gateway/manager/roundtripper/roundtripper.go

```diff
@@ -64,17 +67,14 @@ func (rt *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 		return rt.unauthorizedRT.RoundTrip(req)
 	}
 
-	// No we are going to use token based auth only, so we are reassigning the headers
+	req = utilnet.CloneRequest(req)
 	req.Header.Del("Authorization")
-	req.Header.Set("Authorization", "Bearer "+token)
 
 	if !rt.appCfg.Gateway.ShouldImpersonate {
 		rt.log.Debug().Str("path", req.URL.Path).Msg("Using bearer token authentication")
-
-		return rt.adminRT.RoundTrip(req)
```

Comment:
We used adminRT here and it worked with any token passed since adminRT works without tokens.

Reply:
I also tried to replace adminRT with BearerRT below in the impersonation section, but I got the following error:
```
"level":"error","service":"crdGateway","operation":"get","group":"core.platform-mesh.io","version":"v1alpha1","kind":"AccountInfo","error":"users \"a.shcherbatiuk@sap.com\" is forbidden: User \"a.shcherbatiuk@sap.com\" cannot impersonate resource \"users\" in API group \"\" at the cluster scope: access denied\nNoOpinion"
```
I guess we should setup fga first, let me know if this we should replace adminRT in the impersonation branch as well.